src/lib/messenger.ts

```diff
@@ -12,17 +12,45 @@ type StandardMessage<Type = string, Data = unknown> = {
 	value: Data;
 };
 
+// type IframeIdMessageType = StandardMessage<string, string>;
+
 type ResizeMessage = StandardMessage<
 	'set-ad-height',
-	{ width: number; height: number }
+	{ width?: number | string; height?: number | string }
 >;
 type StringMessage = StandardMessage<'message', string>;
 
-type Message = ResizeMessage | StringMessage;
+type BackgroundMessage = StandardMessage<
+	'background',
+	{
+		scrollType: string;
+		backgroundImage: string;
+		backgroundRepeat: string;
+		backgroundPosition: string;
+		backgroundSize: string;
+		ctaUrl: string;
+	}
+>;
+
+type Message = ResizeMessage | StringMessage | BackgroundMessage;
+
+const generateId = () => {
+	const _4chars = () =>
+		Math.floor((1 + Math.random()) * 0x10000)
+			.toString(16)
+			.substring(1);
+	return `${_4chars()}${_4chars()}-${_4chars()}-${_4chars()}-${_4chars()}-${_4chars()}${_4chars()}${_4chars()}`;
+};
 
 const post = (arg: Message): void => {
-	window.top?.postMessage({ id: self.name, ...arg }, '/');
+	//  frontend messenger.ts discards messages that are not strings and that do not provide an an id in the format of a UUID
+	window.top?.postMessage(JSON.stringify({ id: generateId(), ...arg }), '*');
```

Comment:
pretty sure the `'*'` is safe?

Reply:
Yep we do the same thing in Frontend:

https://github.com/guardian/frontend/blob/ebf8b41a14fe7641aa1fa5e49d65b4908cf8c862/static/src/javascripts/projects/commercial/modules/messenger/post-message.ts#L4